.github/workflows/profiling.yaml

```
@@ -0,0 +1,101 @@
+
+    # Set environment variables for SHAs
+    - name: Set environment variables
+      run: |
+        echo "BASE_SHA=${{ github.event.pull_request.base.sha }}" >> $GITHUB_ENV
+        echo "HEAD_SHA=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+
+    # Run profiler on the base branch
+    - name: Run profiler on base branch
+      env:
+        BASE_SHA: ${{ env.BASE_SHA }}
+      run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:3:14: Double quote to prevent globbing and word splitting

(shellcheck)